### PR TITLE
Clarifications on su and sudo

### PR DESCRIPTION
**Pull request by mark-i-m**
- Added bash _quick substitution_
- Added some improvements to the bullet on `sudo`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,7 +190,7 @@ Notes:
 - For a simple web server for all files in the current directory (and subdirs), available to anyone on your network, use:
 `python -m SimpleHTTPServer 7777` (for port 7777 and Python 2) and `python -m http.server 7777` (for port 7777 and Python 3).
 
-- For running a command with privileges, use `sudo` (for root) or `sudo -u` (for another user). Use `su` or `sudo bash` to actually run a shell as that user. Use `su -` to simulate a fresh login as root or another user.
+- For running a command with privileges, use `sudo` (for root) or `sudo -u` (for another user). Use `sudo bash` to actually run a shell as that user. Use `su` to login as another user and `sudo -i` to login as root. Use `su -` to simulate a fresh login as root or another user.
```
Comment (neil-greenwood) [inline README.md]: Not sure about this change.
`su` will login as root. Maybe you meant `su - user`?
You can also do `sudo -i -u` to login as another user.

Comment (mark-i-m) [inline README.md]: Hmm... I wasn't very clear... Let me try again...

**Comment by mark-i-m** [comment]
Ok, I have tried to improve it. Please let me know what you think

File: README.md [hunk 2]
```diff
@@ -185,7 +190,11 @@ Notes:
 - For a simple web server for all files in the current directory (and subdirs), available to anyone on your network, use:
 `python -m SimpleHTTPServer 7777` (for port 7777 and Python 2) and `python -m http.server 7777` (for port 7777 and Python 3).
 
-- For running a command with privileges, use `sudo` (for root) or `sudo -u` (for another user). Use `su` or `sudo bash` to actually run a shell as that user. Use `su -` to simulate a fresh login as root or another user.
+- For running a command as another user, use `sudo`. Defaults to running as root; use `-u` to specify another user. Use `-i` to login as that user (you will be asked for _your_ password).
+
+- For switching to another user, use `su`. Add `-` to get an environment as if the user just logged in. By default this switches to root, but you can specify a user: `su - user`. You will be asked for the password _of the user you are switching to_.
+
+- For running a command with privileges, use `sudo` (for root) or `sudo -u` (for another user). Use `sudo bash` to actually run a shell as that user. Use `su` to login as another user and `sudo -i` to login as root. Use `su -` to simulate a fresh login as root or another user.
```
Comment (neil-greenwood) [inline README.md]: Almost! You left this incorrect line in-place. :)

**Comment by mark-i-m** [comment]
:facepalm: Fixed thanks :)

**Comment by neil-greenwood** [comment]
👍

File: README.md [hunk 3]
```diff
@@ -185,7 +190,9 @@ Notes:
 - For a simple web server for all files in the current directory (and subdirs), available to anyone on your network, use:
 `python -m SimpleHTTPServer 7777` (for port 7777 and Python 2) and `python -m http.server 7777` (for port 7777 and Python 3).
 
-- For running a command with privileges, use `sudo` (for root) or `sudo -u` (for another user). Use `su` or `sudo bash` to actually run a shell as that user. Use `su -` to simulate a fresh login as root or another user.
+- For running a command as another user, use `sudo`. Defaults to running as root; use `-u` to specify another user. Use `-i` to login as that user (you will be asked for _your_ password).
+
+- For switching to another user, use `su`. Add `-` to get an environment as if the user just logged in. By default this switches to root, but you can specify a user: `su - user`. You will be asked for the password _of the user you are switching to_.
```
Comment (jlevy) [inline README.md]: Minor, but how about we spell it out to avoid confusion:

    For switching the shell to another user, use `su username` or `su - username`. Include the `-` to get an environment as if the user just logged in. Omitting the username defaults to root. You will be asked for the password _of the user you are switching to_.

Comment (mark-i-m) [inline README.md]: +1

**Comment by jlevy** [comment]
Thanks @mark-i-m @neil-greenwood !

I like the expansion of sudo and su, very helpful (just one further note).

Could we remove the ^x^y tip for now? I think it could be covered but it's a bit verbose (with the full example as it stands) and it's a broader topic. In particular, ^x^y only substitutes the first match, which is a gotcha. I think if we do cover it we should give the full set of examples like `!!:gs/x/y/` (which is often better).

**Comment by jlevy** [comment]
Added #453 to track separately.

If you can make those changes glad to merge!

**Comment by mark-i-m** [comment]
@jlevy @neil-greenwood Thanks! I made the suggested changes

**Comment by jlevy** [comment]
👍 Thanks @mark-i-m @neil-greenwood
